code/modules/power/rtg.dm

```diff
@@ -103,4 +104,40 @@
 	circuit = null
 
 /obj/machinery/power/rtg/debug/RefreshParts()
-	return
+	return ..()
+
+/obj/machinery/power/rtg/lavaland
+	name = "Lava powered RTG"
+	desc = "This device only works when exposed to the toxic fumes of Lavaland"
+	circuit = null
+	power_gen = 1500
+
+/obj/machinery/power/rtg/lavaland/Initialize(mapload)
+	. = ..()
+	var/turf/our_turf = get_turf(src)
+	if(!islava(our_turf))
+		power_gen = 0
+	if(!is_mining_level(z))
+		power_gen = 0
```

Comment:
this is horrible it doesnt even prevent you from moving it elsewhere, are you sure this is even laveproof?

Reply:
rtg can't be moved at all, only deconstructed. i'll add a lavaproof flag tho

Reply:
and an anchored for good measure